code-challenges.test.js

```diff
@@ -15,42 +15,165 @@
+// On each iteration set the value.toLowerCase() and check if it .includes() the letter paramater letter.toLowerCase()
+// This will return the strings that include the passed in letter
+// Outside of jest testing console.log the function and paramaters to see the result in the terminal
+
+const matchLetter = (arr,letter) =>{
+     if(typeof letter !== 'string'){
+        return `${letter} is not a letter`
+     }else{
+        return arr.filter(v =>v.toLowerCase().includes(letter.toLowerCase()))
+     }
+    }
+
+
+//  PASS  ./code-challenges.test.js
+//   codedMsg
+//   ✓ takes in a string and returns a coded message. (2 ms)
+//   ✓ takes in a string and returns a coded message.
+
+
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
+describe("fullHouse",() => {
+
+    it("takes in a string and returns a coded message.", () => {
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
+       
+        const hand1 = [5, 5, 5, 3, 3]
+        // Expected output: true
+        const hand2 = [5, 5, 3, 3, 4]
+        // Expected output: false
+        const hand3 = [5, 5, 5, 5, 4]
+        // Expected output: false
 
+      expect(fullHouse([5, 5, 5, 3, 3])).toEqual(true)
+      expect(fullHouse([5, 5, 3, 3, 4])).toEqual(false)
+      expect(fullHouse([5, 5, 5, 5, 4])).toEqual(false)
+        })
+    })
+
+
+// ReferenceError: fullHouse is not defined
 
 // b) Create the function that makes the test pass.
+// Pseudo---->
+// Create a function named fullHouse
+// Pass in an array paramater named arr
+// Create a new variable named newArr 
+// Filter over the arr paramater and create a conditional that compares the index of the current item in the array to the current index of the iteration returning values of the item that do not match its own index.
+// create an if conditional statement that compares the newArr.length strictly equal to three.
+// If true create an if statement that compares the data at the indexes of the new array. If 2 match and one does not then return true. If they do not then return false.
+// if the length is not three then return false in an else statement at the end of the first if statement.
+// Outside of jest testing console.log the function and pass in the array to see the result in the terminal.
+
+const fullHouse =(arr)=> {
+let newArr = arr.filter((item, index) => arr.indexOf(item) !== index)
+    if(newArr.length === 3){
+        if((newArr[0]===newArr[1]&&newArr[0]!==newArr[2]) || (newArr[0]===newArr[2]&&newArr[0]!==newArr[1])){
+            return true
+        }else{
+            return false
+        }
+    }else{
+        return false
+    }
+}
```

Comment:
Something to consider, what would happen if we tested the array [5, 5, 3, 3, 3]?  This should return true, but it looks like it may currently return false with your function.